tfhe/src/integer/server_key/radix_parallel/scalar_comparison.rs

```diff
@@ -0,0 +1,785 @@
+        rhs: T,
+    ) -> RadixCiphertext<PBSOrder>
+    where
+        PBSOrder: PBSOrderMarker,
+        T: DecomposableInto<u8>,
+    {
+        debug_assert!(lhs.block_carries_are_empty());
+
+        let message_modulus = self.key.message_modulus.0;
+        let carry_modulus = self.key.carry_modulus.0;
+        let total_modulus = message_modulus * carry_modulus;
+        let max_value = total_modulus - 1;
+
+        assert!(carry_modulus >= message_modulus);
+        assert!(max_value <= u8::MAX as usize);
+
+        let num_blocks = lhs.blocks.len();
+        let num_blocks_halved = (num_blocks / 2) + (num_blocks % 2);
+
+        let mut scalar_blocks =
+            BlockDecomposer::with_early_stop_at_zero(rhs, total_modulus.ilog2())
+                .iter_as::<u8>()
+                .collect::<Vec<_>>();
+
+        // If we have more scalar blocks than lhs.blocks
+        // and that any of these block additional blocks is != 0
+        // then lhs != rhs
+        let is_scalar_obviously_bigger = scalar_blocks
+            .get(num_blocks_halved..) // We may have less scalar blocks
+            .map(|sub_slice| sub_slice.iter().any(|&scalar_block| scalar_block != 0))
+            .unwrap_or(false);
+        if is_scalar_obviously_bigger {
+            return self.create_trivial_zero_radix(num_blocks);
+        }
+        // If we are sill here, that means scalar_blocks above
+        // num_blocks_halved are 0s, we can remove them
+        // as we will handle them separately.
+        // (truncate can be called even if scalar_blocks.len() < num_blocks_halved);
+        scalar_blocks.truncate(num_blocks_halved);
+
+        let scalar_comp_luts =
+            self.create_scalar_comparison_luts(&scalar_blocks, total_modulus, |x, y| x == y);
+
+        // scalar_blocks.len() is known to be <= to num_blocks_halved
+        // but num_blocks_halved takes into account the non-even num_blocks case
+        let split_index = num_blocks.min(scalar_blocks.len() * 2);
+        let (least_significant_blocks, most_significant_blocks) = lhs.blocks.split_at(split_index);
+
+        let (mut cmp_1, mut cmp_2) = rayon::join(
+            || {
+                let scalar_block_iter = scalar_blocks.into_par_iter();
+
+                let mut packed_blocks = Vec::with_capacity(num_blocks_halved);
+                least_significant_blocks
+                    .par_chunks(2)
+                    .zip(scalar_block_iter)
+                    .map(|(two_blocks, scalar_block)| {
+                        let lut = scalar_comp_luts[scalar_block as usize]
+                            .as_ref()
+                            .expect("internal error, missing scalar comparison lut");
+                        let mut packed_block = self.pack_block_chunk(two_blocks);
+                        self.key.apply_lookup_table_assign(&mut packed_block, lut);
+                        packed_block
+                    })
+                    .collect_into_vec(&mut packed_blocks);
```

Comment:
Yes it allows to reduce the number of binary blocks we have to sum.

On a m6i with 128 threads, in param_2_2, each thread can sum 15 blocks, so packing won't bring improvements
on smaller machines it may bring one.

The main interest of packing is to reduce number of pbs and hopefully leave room for things like multibit pbs to function beter